### DISCUSSION TITLE
how to setup 1 primary and 2 backups?

### DISCUSSION THREAD

**Discussion by tessus** [acassen/keepalived#2261] [Q&A] [answered]
I currently use keepalived for an active-passive pi-hole cluster. It works splendidly.

Out of curiousity I was wondering, whether it was possible to use 2 backups. I've searched the net, but I only found documents on how to setup a simple failover cluster with 2 machines.
None of the documents ever mentioned how to use more than 1 backup, thus more than 2 machines. Let's say the primary fails, then the backup. What then? It's unlikely but this is a hypothetical scenario. I would like that a 2nd backup would then take over.

Is this even possible? I can also post my current config, if required.

**Comment by pqarmitage** [comment, marked as answer]
In theory you can have up to 254 backups. VRRP instances have a priority between 1 and 255, so you can have 255 machines with different priorities for the same virtual router.

So just another machine and ensure that the priority configured in the VRRP instance is different from the two systems you already have.

**Reply by nser77** [reply]
Just remember that priority 255 owns IPvX:

```
[...]

   The priority value for the VRRP router that owns the IPvX address
   associated with the virtual router MUST be 255 (decimal).

   VRRP routers backing up a virtual router MUST use priority values
   between 1-254 (decimal).  The default priority value for VRRP routers
   backing up a virtual router is 100 (decimal).

[...]
````
